pkg/apis/aws/helper/scheme.go

```diff
@@ -44,6 +45,18 @@ func CloudProfileConfigFromCluster(cluster *controller.Cluster) (*api.CloudProfi
 	return cloudProfileConfig, nil
 }
 
+// InfrastructureFromCluster decodes the infrastructure for a shoot cluster
+func InfrastructureFromCluster(cluster *controller.Cluster) (*extensionsv1alpha1.Infrastructure, error) {
+	var infra *extensionsv1alpha1.Infrastructure
+	if cluster != nil && cluster.Shoot != nil && cluster.Shoot.Spec.Provider.InfrastructureConfig != nil && cluster.Shoot.Spec.Provider.InfrastructureConfig.Raw != nil {
+		infra = &extensionsv1alpha1.Infrastructure{}
+		if _, _, err := decoder.Decode(cluster.Shoot.Spec.Provider.InfrastructureConfig.Raw, nil, infra); err != nil {
```

Comment:
How does this work and did you test this at all? It should not be possible to decode `shoot.spec.provider.infrastructureConfig` to `*extensionsv1alpha1.Infrastructure`. `shoot.spec.provider.infrastructureConfig` should be decoded to `*api.InfrastructureConfig` instead.